src/components/Services.jsx

```diff
@@ -11,50 +12,52 @@ import KonnectorTile from 'components/KonnectorTile'
 import classNames from 'classnames'
 import fillWithGhostItems from 'components/helpers/fillWithGhostItems'
 import { getInstalledKonnectors } from 'reducers/index'
+import useAppsInMaintenance from 'hooks/withAppsInMaintenance'
 
-class Services extends Component {
-  render() {
-    const { t, installedKonnectors } = this.props
-    const hasConnections = !!installedKonnectors.length
-    return (
-      <div
-        className={classNames('services-list-wrapper', {
-          'services-list-wrapper--empty': !hasConnections
-        })}
-      >
-        {hasConnections ? (
-          <div className="list connector-list" data-tutorial="home-services">
-            {installedKonnectors.map((konnector, index) => (
-              <KonnectorTile
-                key={index}
-                konnector={konnector}
-                route={`connected/${konnector.slug}`}
-              />
-            ))}
-            {<AddServiceTile label={t('add_service')} />}
-            {fillWithGhostItems(6)}
+const Services = ({ t, installedKonnectors, client }) => {
+  const hasConnections = !!installedKonnectors.length
+  const appsInMaintenance = useAppsInMaintenance(client)
+  return (
+    <div
+      className={classNames('services-list-wrapper', {
+        'services-list-wrapper--empty': !hasConnections
+      })}
+    >
+      {hasConnections ? (
+        <div className="list connector-list" data-tutorial="home-services">
+          {installedKonnectors.map((konnector, index) => (
+            <KonnectorTile
+              key={index}
+              konnector={konnector}
+              route={`connected/${konnector.slug}`}
+              isInMaintenance={appsInMaintenance.some(
+                app => app.slug === konnector.slug
+              )}
```

Comment:
At each konnector tile, we scan all the appsInMaintenance array to know if the app is in maintenance, which makes the rendering O(n^2). What do you think of using lodash's keyBy a bit higher to build an object so that knowing if an app is in maintenance is in O(1) ?

Reply:
Yup, sounds good!